pkg/secretsmanager/secretsmanager.go

```diff
@@ -73,54 +76,27 @@ type secretManagerNone struct {
 // NewSecretManager creates a new SecretManager object
 func NewSecretManager(ctx context.Context, instance *v1alpha1.SecretAgentConfiguration, cloudCredNS string, rClient client.Client) (context.Context, SecretManager, error) {
 
+	// get namespace if not previously deployed
 	config := &instance.Spec.AppConfig
 	if len(cloudCredNS) == 0 {
 		cloudCredNS = instance.Namespace
 	}
 
-	if instance.Spec.AppConfig.CredentialsSecretName != "" {
-		// load credentials secret
-		secObject, err := k8ssecrets.LoadSecret(rClient,
-			config.CredentialsSecretName, cloudCredNS)
-		if err != nil {
-			log.Error(err, "error loading cloud credentials secret from the Kubernetes API",
-				"secret_name", config.CredentialsSecretName,
-				"cloud_secret_namespace", cloudCredNS)
-			return ctx, nil, err
-		}
-
-		var dir string
-
-		if err := manageCloudCredentials(config.SecretsManager, secObject, dir); err != nil {
-			log.Error(err, "error loading cloud credentials from secret provided",
-				"secret_name", config.CredentialsSecretName,
-				"cloud_secret_namespace", cloudCredNS)
-		}
-	}
-
 	var sm SecretManager
 	var err error
 
 	// decide which SecretManager type based on AppConfig
 	switch config.SecretsManager {
 	case v1alpha1.SecretsManagerGCP:
-		sm, err = newGCP(ctx, config)
+		sm, err = newGCP(ctx, config, rClient, cloudCredNS)
 		if err != nil {
 			log.Error(err, "couldn't create a new GCP object")
 			return ctx, nil, err
 		}
-		dir, err := ioutil.TempDir("", "cloud_credentials-*")
-		println(dir)
-		if err != nil {
-			log.Error(err, "couldn't create a temporary credentials dir")
-			return ctx, nil, err
-		}
-		// clean up after ourselves
-		defer os.RemoveAll(dir)
 	case v1alpha1.SecretsManagerAWS:
-		ctx, sm = newAWS(ctx, config)
+		ctx, sm, err = newAWS(ctx, config, rClient, cloudCredNS)
```

Comment:
I realize this is from the last PR but what are we returning the context here?

Reply:
Because the ctx gets used in the client creation then passed back down the controller.  Apparently it has to be passed through all functions. Can't be used in a struct.